### DISCUSSION TITLE
Terraform modules

### DISCUSSION THREAD

**Discussion by lmal54321** [aquasecurity/trivy#6134] [False Detection]
### IDs

AVD-AWS-0130 / Terraform module interpolation

### Description

`module "example" {
  source  = "terraform-aws-modules/autoscaling/aws"
  version = "~> 6.0"
  name    = "trivy"

  min_size                  = 1
  max_size                  = 1
  desired_capacity          = 1
  wait_for_capacity_timeout = 0
  health_check_type         = "EC2"
  vpc_zone_identifier       = "subnet"

  metadata_options = {
    http_endpoint = "enabled"
    http_tokens   = "required"
  }

  image_id                             = "someami"
  instance_type                        = t2.micro
}`

When calling a module which has a bad default - it flags as a false positive even though I am passing the variable to the module which corrects this.

It would be good if it parsed the interpolated values and flags it on module.example.resource if it was incorrect. Currently it flags on resource.aws_launch_template.this even though metadata_options is set correctly.

This isn't exclusive to this - but a behaviour in modules vs resources.

### Reproduction Steps

```bash
1. create a main.tf file
2. trivy config --format json --severity HIGH .
```

### Target

Filesystem

### Scanner

Misconfiguration

### Debug Output

```bash
"Results": [
    {
      "Target": ".",
      "Class": "config",
      "Type": "terraform"
    },
    {
      "Target": "terraform-aws-modules/autoscaling/aws/main.tf",
      "Class": "config",
      "Type": "terraform",
      "MisconfSummary": {
        "Successes": 0,
        "Failures": 1,
        "Exceptions": 0
      },
      "Misconfigurations": [
        {
          "Type": "Terraform Security Check",
          "ID": "AVD-AWS-0130",
          "AVDID": "AVD-AWS-0130",
          "Title": "aws_instance should activate session tokens for Instance Metadata Service.",
          "Description": "\nIMDS v2 (Instance Metadata Service) introduced session authentication tokens which improve security when talking to IMDS.\nBy default \u003ccode\u003eaws_instance\u003c/code\u003e resource sets IMDS session auth tokens to be optional. \nTo fully protect IMDS you need to enable session tokens by using \u003ccode\u003emetadata_options\u003c/code\u003e block and its \u003ccode\u003ehttp_tokens\u003c/code\u003e variable set to \u003ccode\u003erequired\u003c/code\u003e.\n",
          "Message": "Launch template does not require IMDS access to require a token",
          "Query": "data..",
          "Resolution": "Enable HTTP token requirement for IMDS",
          "Severity": "HIGH",
          "PrimaryURL": "https://avd.aquasec.com/misconfig/avd-aws-0130",
          "References": [
            "https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service",
            "https://avd.aquasec.com/misconfig/avd-aws-0130"
          ],
          "Status": "FAIL",
          "Layer": {},
          "CauseMetadata": {
            "Resource": "module.example",
            "Provider": "AWS",
            "Service": "ec2",
            "StartLine": 21,
            "EndLine": 236,
            "Code": {
              "Lines": [
                {
                  "Number": 21,
                  "Content": "resource \"aws_launch_template\" \"this\" {",
                  "IsCause": true,
                  "Annotation": "",
```

### Version

```bash
0.49.1
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by simar7** [comment]
Trivy can override templates with terraform variables if you have them defined within your terraform code with the `variable` block. In your example that you've shared I didn't see a variable definition. See for details https://developer.hashicorp.com/terraform/language/values/variables

You can also define them elsewhere and override by passing a variable file to Trivy via the `--tf-vars` flag.

**Comment by lmal54321** [comment]
The override is there though:

metadata_options = {
http_endpoint = "enabled"
http_tokens = "required"
}

This is what gets passed to the module.

Are you saying I should have metadata_options = var.somevar ? I don't think that would help and isn't the standard.

**Comment by nikpivkin** [comment]
@lmal54321 Hi!

This issue is related to dynamic blocks. Track https://github.com/aquasecurity/trivy/issues/5902
